x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/action/TransportDeleteRollupJobAction.java

```diff
@@ -5,103 +5,96 @@
+                                 ActionListener<DeleteRollupJobAction.Response> listener) {
 
-                // Step 2. Wait for the task to finish cancellation internally
-                persistentTasksService.waitForPersistentTaskCondition(jobId, Objects::isNull, timeout,
-                        new PersistentTasksService.WaitForPersistentTaskListener<RollupJob>() {
-                            @Override
-                            public void onResponse(PersistentTasksCustomMetaData.PersistentTask<RollupJob> task) {
-                                logger.debug("Task for Rollup job [" + jobId + "] successfully canceled.");
-                                listener.onResponse(new AcknowledgedResponse(true));
-                            }
+        assert jobTask.getConfig().getId().equals(request.getId());
 
-                            @Override
-                            public void onFailure(Exception e) {
-                                logger.error("Error while cancelling task for Rollup job [" + jobId
-                                        + "]." + e);
-                                listener.onFailure(e);
-                            }
-
-                            @Override
-                            public void onTimeout(TimeValue timeout) {
-                                String msg = "Stopping of Rollup job [" + jobId + "] timed out after [" + timeout + "].";
-                                logger.warn(msg);
-                                listener.onFailure(new ElasticsearchException(msg));
-                            }
-                        });
-            }
-
-            @Override
-            public void onFailure(Exception e) {
-                logger.error("Error while requesting to cancel task for Rollup job [" + jobId
-                        + "]" + e);
-                listener.onFailure(e);
-            }
-        });
+        // Tell the task to shut down.  TODO if the indexer is running this just signals to shutdown.
+        // The actual shutdown may be delayed, which we can fix with an await in a followup PR
+        jobTask.onCancelled();
+        listener.onResponse(new DeleteRollupJobAction.Response(true));
+    }
 
+    @Override
+    protected DeleteRollupJobAction.Response newResponse(DeleteRollupJobAction.Request request, List<DeleteRollupJobAction.Response> tasks,
+                                                       List<TaskOperationFailure> taskOperationFailures,
+                                                       List<FailedNodeException> failedNodeExceptions) {
+        // There should theoretically only be one task running the rollup job
+        // If there are more, in production it should be ok as long as they are acknowledge shutting down.
+        // But in testing we'd like to know there were more than one hence the assert
+        assert tasks.size() == 1;
+        boolean cancelled = tasks.stream().anyMatch(DeleteRollupJobAction.Response::isDeleted);
```

Comment:
Given the comment above should this not look for all tasks in the stream being deleted rather than just one?